libr/core/cmd_anal.c

```diff
@@ -2779,6 +2780,24 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 			r_core_cmdf (core, "\"afs %s\"", data);
 			free (sig);
 			free (data);
+		} else if (input[2] == 'r') { // "afsr"
+			if (input[3] == ' ') {
+				const char *ntype = (input[4])
+					? input + 4
+					: NULL;
+				if (ntype) {
+					char *sig = r_core_cmd_str (core, "afs");
+					char *sig_notype = strchr (sig, '('); 
+					strtok (sig_notype, "\n");
+					sig_notype -= 2;
```

Comment:
i think using r_str_rchr (reverse chr search) will be better and simpler than doing all that